tests/gates/snapshots/approximate_test/test_approximate_rz_with_compression/phase1/gates.txt

```diff
@@ -1 +1 @@
-[]
\ No newline at end of file
+[S(target_qubit=1)]
```

Comment:
Shouldn't this be `[PauliRotations(target_qubit=1, phase=Fraction(1, 2), axis=Z)]`? maybe we do have a double call to `to_clifford_plus_t`

Reply:
It's not a double call! The logic in `approximate.py`: calls `from_gate_string` only if the length of the partition is larger than 1. If it is just one, it uses the same code as the `compress_rotations = False` case. Giving out H, S and T gates respectively

Reply:
Made changes in the latest push and modified the behavior! The logic was that it was passing to the `from_gate_string` only if the string was longer than length 1! Another question, should the single Hadamards also be converted to a rotation! Right now, I leave them as it is.

Reply:
Oh I see, if it was intentional there is no problem. Hadamards can stay hadamards because we can do them transversally :+1: